src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -317,7 +319,26 @@ public void respawn()
 	@Override
 	public void setGameMode(@NotNull GameMode mode)
 	{
-		gamemode = mode;
+		PlayerGameModeChangeEvent gameModeChangeEvent = new PlayerGameModeChangeEvent(this, mode);
+
+		getServer().getPluginManager().callEvent(gameModeChangeEvent);
+
+		if (!gameModeChangeEvent.isCancelled())
+		{
+			gamemode = mode;
+		}
```

Comment:
setGameMode and simulateChangeGameMode are actually the same, so it would be nice to use
```suggestion
		simulateChangeGameMode(mode);
```
inside setGameMode() method instead copypasting the code

Reply:
Adjusted, good point was little bit like "I mean it works" mentality xD